README.md

```diff
@@ -9,6 +9,90 @@ go get github.com/sourcegraph/sourcegraph-accounts-sdk-go
 > [!note]
 > Please submit all issues to the [`sourcegraph/sourcegraph-accounts` repository](https://github.com/sourcegraph/sourcegraph-accounts/issues) 
 
+## Authentication
+
+The following example demonstrates how to use the SDK to set up user authentication flow with SAMS for your service.
+
+In particular,
+
+- The route `/auth/login` is where the user should be redirected to start a new authentication flow.
+- The route `/auth/callback` is where the user will be redirected back to the service after completing the authentication on the SAMS side.
+
+```go
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+
+	samsauth "github.com/sourcegraph/sourcegraph-accounts-sdk-go/auth"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+)
+
+type stateStore struct{
+	// Authentication state is the unique identifier that is randomly-generated and
+	// assigned to a particular authentication flow, they are used to prevent
+	// authentication interception attacks and considered secrets, therefore it MUST
+	// be stored in a backend component (e.g. Redis, database). The design of the
+	// samsauth.StateStore interface explicitly disallowed storing state in the
+	// cookie, as they can be tampered with when cookie values are stored
+	// unencrypted.
+}
+
+func (s *stateStore) SetState(r *http.Request, state string) error {
+	// TODO: Save to session data.
+	return nil
+}
+
+func (s *stateStore) GetState(r *http.Request) (string, error) {
+	// TODO: Retrieve from session data.
+	return "", nil
+}
+
+func (s *stateStore) DeleteState(r *http.Request) {
+	// TODO: Delete from session data.
+}
+
+func main() {
+	samsauthHandler, err := samsauth.NewHandler(
+		samsauth.Config{
+			Issuer:         "https://accounts.sourcegraph.com",
+			ClientID:       os.Getenv("SAMS_CLIENT_ID"),
+			ClientSecret:   os.Getenv("SAMS_CLIENT_SECRET"),
+			// RequestScopes needs to include all the scopes that the service needs to
+			// access on behalf of the user. Scopes that are only used for Clients API are
+			// not needed here.
+			RequestScopes:  []scopes.Scope{scopes.OpenID, scopes.Email, scopes.Profile},
+			RedirectURI:    os.Getenv("SAMS_REDIRECT_URI"),
+			FailureHandler: samsauth.DefaultFailureHandler,
+			StateStore:     &stateStore{},
+		},
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/auth/login", samsauthHandler.LoginHandler())
+	mux.Handle("/auth/callback", samsauthHandler.CallbackHandler(
```

Comment:
Do we need to enforce the allowed HTTP methods of these routes?

Reply:
Ahh, the curse of example code haha, the stdlib mux doesn't provide method selector, but ultimately, I think there is no reason why we _have to_ be strict about it.